features/FEATURE_BLE/targets/TARGET_CORDIO/driver/CordioHCIDriver.h

```diff
@@ -58,6 +58,10 @@ struct buf_pool_desc_t {
  *   - Access to the write function of the underlying HCITransport driver.
  */
 class CordioHCIDriver {
+
+    // hook for internal tests and passthrough driver
+    friend class CordioHCIHook;
```

Comment:
not that I mind but why not just set the handler public?

Reply:
Encapsulation. 

If it was public then it means that the system has been designed to support replacement of the RX handler; it hasn't and it won't! This would lead to disasters if application code was replacing the RX handler.